include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2419,28 +2419,44 @@ namespace internal
 
       // re-orientation
       std::array<const unsigned int *, n_face_orientations> orientation;
-      orientation[0] = (data.data.front().nodal_at_cell_boundaries == true) ?
-                         &data.face_orientations[face_orientation[0]][0] :
-                         &dummy;
+      if (n_face_orientations == 1)
+        orientation[0] = (data.data.front().nodal_at_cell_boundaries == true) ?
+                           &data.face_orientations[face_orientation[0]][0] :
+                           &dummy;
+      else
+        {
+          for (unsigned int v = 0; v < VectorizedArrayType::size(); ++v)
+            {
+              if (cells[v] == numbers::invalid_unsigned_int)
+                continue;
+
+              orientation[v] =
+                (data.data.front().nodal_at_cell_boundaries == true) ?
+                  &data.face_orientations[face_orientation[v]][0] :
+                  &dummy;
+            }
+        }
 
       // face_to_cell_index_hermite
       std::array<const unsigned int *, n_face_orientations> index_array_hermite;
 
-      index_array_hermite[0] =
-        (data.data.front().nodal_at_cell_boundaries == true && fe_degree > 1 &&
-         data.element_type == MatrixFreeFunctions::tensor_symmetric_hermite) ?
-          &data.face_to_cell_index_hermite(face_no[0], 0) :
-          &dummy;
+      if (n_face_orientations == 1)
+        index_array_hermite[0] =
+          (data.data.front().nodal_at_cell_boundaries == true &&
+           fe_degree > 1 &&
+           data.element_type == MatrixFreeFunctions::tensor_symmetric_hermite) ?
+            &data.face_to_cell_index_hermite(face_no[0], 0) :
+            &dummy;
 
       if (n_face_orientations > 1 &&
           data.data.front().nodal_at_cell_boundaries == true && fe_degree > 1 &&
           data.element_type == MatrixFreeFunctions::tensor_symmetric_hermite)
         {
-          const unsigned int n_filled_lanes =
-            dof_info.n_vectorization_lanes_filled[dof_access_index][cell];
-
-          for (unsigned int v = 1; v < n_filled_lanes; ++v)
+          for (unsigned int v = 0; v < VectorizedArrayType::size(); ++v)
             {
+              if (cells[v] == numbers::invalid_unsigned_int)
+                continue;
```

Comment:
Just to help understanding, would you mind adding a comment that the loop breaks once an `invalid_unsigned_int` is hit for all cases except the exterior faces in the ECL loop (where some faces might be at the boundaries but others not)?